include/blas_meta.h

```diff
@@ -120,6 +120,11 @@ static inline index_t get_power_of_two(index_t wGSize, bool rounUp) {
 
 #define SYCL_BLAS_INLINE inline __attribute__((always_inline))
 
+template<typename index_t>
+static  SYCL_BLAS_INLINE index_t roundUp(index_t x, index_t y) {
+  return ((((x) + (y)-1) / (y)) * (y));
```

Comment:
Mehdi copied this code from somewhere, either another project or an older version of this one.

It's used in calculating the global size for the non-shared-memory version of the kernel. But since the user cannot specify which version to use, I made shared-memory the default and left the other global size calculation commented out. I don't know if I should remove it, but if we want to add the possibility of using the non-shared-memory kernel in the future then this function would be used and the commented out code in the interface file.

Reply:
Yes this code looks very much like it has been copied from a macro because parentheses are useful in that case, because of operator priorities. Here it is not necessary.

Concerning the fact that the function is not used yet, I would comment it out but some people don't like to comment out code so I'm not sure what to do about it

Reply:
yes it was copied from macro, since you are using the function, there is no need for prantheses